backend/capellacollab/projects/toolmodels/modelsources/git/github/handler.py

```diff
@@ -131,6 +140,25 @@ def get_artifact_from_job(
             artifact_response, trusted_path_to_artifact
         )
 
+    def get_last_commit_id_for_file_path(
+        self, project_id: str, file_path: str, revision: str | None
+    ) -> t.Any:
+        if revision:
+            attribute = f"rev={revision}"
+        else:
+            attribute = f"path={file_path}&rev={self.git_model.revision}"
+        response = requests.get(
+            f"{self.git_instance.api_url}/repos{project_id}/commits?{attribute}",
+            headers=self.__get_headers(self.git_model.password)
+            if self.git_model.password
+            else "",
```

Comment:
```suggestion
            else None,
```

Reply:
as a heads up: that is a weird thing in the github api: usually it was totally fine if just the token was None but in that specific case if the token was none it worked as not authenticated while no headers at all worked fine... so e.g. in `__get_file_from_repository` it is ok if the token is None but not here

Reply:
But you're setting the headers to "", not the token.